### PR TITLE
Log Analytics workspace key should be retrieved and set by fake RP

### PR DESCRIPTION
**Pull request by charlesakalugwu**
Currently, the ARO plugin is fetching the workspace key. This will not work in the RP environment because the plugin has a service principal with access to the publisher tenant, but the Log analytics workspace is not in that tenant. As a result, the plugin will get a 404 or auth error.  

The workspace key retrieval logic should not have gone into the plugin as per the design doc for this feature. Workspace key retrieval should be handled by the RP (fake RP in our case).

```release-note
Log analytics workspace key should not be retrieved via plugin
```

depends on #1931 

/cc @asalkeld @mjudeikis

### CODE REVIEW

**Comment by charlesakalugwu** [comment]
e2e-create flake #1778 
artifacts: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/1943/pull-ci-azure-master-e2e-create/1083/build-log.txt

Noticed that the sync pod constantly updates the log analytics agent deployment and daemonset. 
*log-analytics-node-agent (daemonset)*
- sync wants to set spec.containers[0].securityContext.runAsGroup = 0
- sync wants to apiVersion = v1 on the `valueFrom.fieldRef` in the env variables section of the yaml

*log-analytics-cluster-agent (deployment)*
- sync wants to set spec.replicas = 1
- sync wants to set spec.containers[0].securityContext.runAsGroup = 0
- sync wants to apiVersion = v1 on the `valueFrom.fieldRef` in the env variables section of the yaml

These issues are fixed by updating those resource yamls to set those values to make them invariants which prevents constant reconciliation of those resources by the sync pod

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -238,21 +235,6 @@ func (p *plugin) createOrUpdateExt(ctx context.Context, cs *api.OpenShiftManaged
...
-		if err != nil {
-			return &api.PluginError{Err: err, Step: api.PluginStepClientCreation}
-		}
-		wc := operationalinsights.NewWorkspacesClient(ctx, p.log, cs.Properties.AzProfile.SubscriptionID, authorizer)
-		if err != nil {
-			return &api.PluginError{Err: err, Step: api.PluginStepClientCreation}
-		}
-		err = enrich.MonitorIDAndKey(ctx, wc, cs)
-		if err != nil {
-			return &api.PluginError{Err: err, Step: api.PluginStepEnrichMonitorProfile}
```
Comment (asalkeld) [inline pkg/plugin/plugin.go]: This "api.PluginStepEnrichMonitorProfile" is not used anywhere else, we should remove it.

Comment (charlesakalugwu) [reply pkg/plugin/plugin.go]: Yes. However that doesn't block this PR.

Comment (mjudeikis) [reply pkg/plugin/plugin.go]: I think they should be removed here too as steps removal is happening here too

**Comment by JackQuincy** [comment]
LGTM

**Pull request merged**